src/utilities/Dealer.cpp

```diff
@@ -19,93 +23,124 @@ Dealer::DealerFlag::DealerFlag(DealerFlagTitle title, DealerFlagPriority priorit
     for (auto const &[roleName, dealerFlags] : flagMap) {
-        orderedRoleNames.push_back(roleName);
+        roleNames.push_back(roleName);
     }
-    unordered_map<string, v::RobotView> result;
-    for (int i = 0; i < orderedRoleNames.size(); i++) {
-        if (assignment[i] != -1) {
-            result.insert({orderedRoleNames[i], allRobots[assignment[i]]});
+
+    // Loop through the order of role priorities (column)
+    for (int i = sizeof(DealerFlagPriority); i >= 0; i--){
+        std::vector<std::vector<double>> currentScores; // Scores to be distributed (highest Priority)
+        std::vector<int> newAssignments;                // Assignments from these scores
+        std::vector<int> currentRoles;                  // Index of roles inside Score (role column)
+        std::vector<int> originalRolesIndex;            // Index of roles from original index (role number)
+        std::vector<int> currentIDs;                    // Index of ID's inside Score (robot row)
+        std::vector<int> originalIDsIndex;              // Index of ID's from original index (robot id)
+
+        // Check if a column has the looked for priorities
+        for (int j = 0; j < scores.size(); j++){
+            // if so, add it to a list and save the index
+            if (scores.at(j).second == i) {
+                currentScores.push_back(scores.at(j).first);    // get the score column
+                currentRoles.push_back(j);                      // get the current index
+                originalRolesIndex.push_back(indexRoles[j]);    // get the role number
+            }
+        }
+        if (currentRoles.size() > 0) {
+            // Return best assignment for those roles (column)
+            rtt::Hungarian::Solve(currentScores, newAssignments);
+            if (newAssignments.size() > 0) {
+                for (int j = 0; j < newAssignments.size(); j++) {
+                    if (newAssignments[j] >= 0) {
+                        currentIDs.push_back(newAssignments[j]);                    // get newly assigned robot from current index
+                        originalIDsIndex.push_back(indexID[currentIDs.back()]);     // get robot number
+                        output.insert({roleNames[originalRolesIndex[j]], allRobots[originalIDsIndex.back()]});
+                    }
+                }
+                if (output.size() == allRobots.size()) return output;               // case if there are less then 11 bots to distribute
+                std::sort(currentIDs.begin(), currentIDs.end());                    // Sort to delete from back to front
+                // Delete assigned roles and robots from score
+                for (int j = currentRoles.size() - 1; j >= 0; j--) {
+                    scores.erase(scores.begin() + currentRoles[j]);                 // remove role from score (col)
+                    indexRoles.erase(indexRoles.begin() + currentRoles[j]);         // remove from index list
+                }
```

Comment:
there's a .clear() 🤔

Reply:
It has to remove only certain elements of the vector, for example: element: erase 7 4 2 from 1 2 3 4 5 6 7 8 makes 1 3 5 6 8.

Reply:
Look at it with Rolf.